flat_mutation_reader.hh

```diff
@@ -131,8 +131,7 @@ public:
         // Stops when consumer returns stop_iteration::yes or end of stream is reached.
         // Next call will start from the next mutation_fragment in the stream.
         future<> consume_pausable(Consumer consumer, db::timeout_clock::time_point timeout) {
-            return do_with(std::move(consumer), [this, timeout] (Consumer& consumer) {
-                return repeat([this, &consumer, timeout] {
+                return repeat([this, consumer = std::move(consumer), timeout] () mutable {
```

Comment:
The consumer can now be moved while it's being called (repeat moves the lambda), while before it couldn't move. Whether that's a problem depends on whether consumers rely on stable `this`. Did you check all consumers?

We should document this property.

Reply:
@tgrabiec prior to my patch: 515bed90bbb762ca1b6cd670566f3af562613f47
the consumer was moved into a `do_until` lambda, similar to this `repeat` lambda,
so in principle it should be safe.  Just need to see if any consumer introduced after 515bed90bbb762ca1b6cd670566f3af562613f47 would have a problem if moved.

Reply:
@haaawk I'd like to commit this series but that's the last hanging question, could you double-check and ack that "no consumer introduced after 515bed9 would have a problem if moved"?

Reply:
All the consumers I've found are ok and don't relay on stable `this`

Reply:
@psarna